src/main.rs

```diff
@@ -41,20 +43,29 @@ fn main() {
     } else {
         println!("\nNot Solvable!");
     }
+
+    Ok(())
 }
 
-fn check_data(num_cols: &mut usize, num_rows: &mut usize, puzzle: &mut Vec<StringRecord>) -> Result<(), Box<dyn Error>> {
+fn check_data(
+    num_cols: &mut usize,
+    num_rows: &mut usize,
+    puzzle: &mut Vec<StringRecord>,
+) -> Result<(), anyhow::Error> {
     let mut rdr = csv::ReaderBuilder::new()
         .has_headers(false)
         .from_reader(io::stdin());
 
-    let first = rdr.records().next().unwrap()?;
+    let first = rdr
+        .records()
+        .next()
+        .ok_or_else(|| anyhow::anyhow!("no records found."))??;
```

Comment:
Unwrap will cause the application to panic, which is unlikely to be what we want.